micro-rdk/src/common/grpc.rs

```diff
@@ -145,6 +145,7 @@ where
     }
 
     // TODO(RSDK-9242): Use the builder pattern instead.
+    #[allow(dead_code)]
```

Comment:
I opted not to attempt to actually avoid compiling the local signaling support when the `local-signaling` feature was off, because it got really messy with the `Either` type over the app signaling vs local signaling. I doubt the code savings are significant. If anyone thinks it would be worth it, I'm happy to file a ticket for sometime down the road to attempt it.

Reply:
does this mean we can't complie if `local-signaling` flag is disabled?

This feels like a weird state to leave things in, and I personally think we should not have a flag for local signaling anyway. If anything, we can have an env var to check if local signaling is disabled and adds/removes it in the builder step.

Reply:
No, this is exactly so we *can* compile / clippy when the `local-signaling` flag is disabled. Otherwise we get warnings here that `register_signaling_server` is unused.

Reply:
gotcha, thanks for clarifying.